### ISSUE TITLE
AOT compiler strips down "children" routes when created dynamically

### ISSUE THREAD

**Issue by jsgoupil** [angular/angular#20249]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

    ng serve --aot

There is a problem with the AOT. It's duplicating the call of my route configuration.
It fails right here:

    throw new Error("Invalid configuration of route '" + fullPath + "'. One of the following must be provided: component, redirectTo, children or loadChildren");
I searched for the token ROUTES and I found this one in the main.bundle.js

    (1024, __WEBPACK_IMPORTED_MODULE_6__angular_router__[\"i\" /* ROUTES */], function () { return [[{ path: \"\", component: __WEBPACK_IMPORTED_MODULE_2__app_component__[\"a\" /* AppComponent */] }, { path: \"areas\" }]]; }, []),

## Expected behavior
It should not trim my route neither call the ROUTES provider more than once?
If I run

    ng serve

it works fine.

## Minimal reproduction of the problem with instructions
I have created a github repo for this.
https://github.com/jsgoupil/router-aot-error

I started from the CLI, then added this commit:
jsgoupil/router-aot-error@3c487ae

## Environment

<pre><code>
Angular version: 5.0.0

 
For Tooling issues:
- Node version: 6.11.3
- Platform:  Windows
</code></pre>

This is a follow up on https://github.com/angular/angular/issues/19336#issuecomment-342565431

**Comment by jasonaden** [comment]
We need to look into why this is happening and make a smaller reproduction of this issue. It's marked as a regression, but it could just be something we don't support. Pulling it into planned work for Dec 19th.

**Comment by alxhub** [comment]
The issue as originally reported is WAI, in the sense that the code is unsupported by AOT. The `makeRoutes` function which ends up being called (defined in `demo-page.ts`) is not statically analyzable.

However, I do believe there are several compiler bugs here:

1) This code:

```
export const routes: Routes = [
].concat(route());
```

generates this metadata:

```
"routes": {
  "__symbolic": "call",
  "arguments": [
    {
      "__symbolic": "call",
      "expression": {
        "__symbolic": "reference",
        "name": "route"
      }
    }
  ]
}
```
This is incorrect - the outer `call` has no expression, which should be a reference to the `concat` method of a new empty array.

2) AOT gives no error that any of these constructions are invalid, and the code is left to fail at runtime.

In short, this is not a regression, but we can do better on the error messages.

**Comment by jsgoupil** [comment]
This is still happening with Angular 6.

**Comment by sgehrman** [comment]
same  problem, can't find the simple solution

**Comment by JoostK** [comment]
Closing as this is no longer an issue in the Ivy compiler.
